pkg/auth/oauth2authenticator.go

```diff
@@ -136,15 +120,45 @@ func createVerifier(count int) []byte {
 	lut := []byte("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~")
 	verifier := make([]byte, count)
 
-	// TODO is this good enough?
 	for i := range verifier {
-		index := rand.Int() % len(lut)
+		index := randIndex(len(lut))
 		verifier[i] = lut[index]
 	}
 
 	return verifier
 }
 
+const maxRandIndexLimit = 256
+
+// randIndex provides a secure random number in the range [0, limit).
+//
+// Rejection sampling and modulo reduction are applied to avoid modulo bias and
+// ensure a uniform distribution.
+//
+// This function panics if the limit exceeds maxRandIndexLimit, or if a secure
+// random source cannot be read.
+//
+// Further reading:
+// https://research.kudelskisecurity.com/2020/07/28/the-definitive-guide-to-modulo-bias-and-how-to-avoid-it/
+// https://stackoverflow.com/a/10984975
+func randIndex(limit int) int {
+	if limit > maxRandIndexLimit {
+		panic(fmt.Sprintf("unsupported randIndex limit %d", limit))
+	}
+	rejectLimit := 256 - (256 % limit)
+	for {
+		var buf [1]byte
+		_, err := rand.Reader.Read(buf[:])
+		if err != nil {
+			panic(fmt.Sprintf("failed to read secure random bytes: %v", err))
+		}
+		r := int(buf[0])
```

Comment:
Question: Is there a reason we use only byte 0? If we would make use of the full integer siye, I assume we wouldn't need to limit this to 256, or am I missing something?